woodwork/serialize.py

```diff
@@ -207,5 +207,9 @@ def _create_archive(tmpdir):
 
 def save_orc_file(dataframe, filepath):
     from pyarrow import Table, orc
-    pa_table = Table.from_pandas(dataframe, preserve_index=False)
+    df = dataframe.copy()
```

Comment:
We should be careful with copy operations (especially with large datasets). Is there a way to avoid this?

Reply:
I had the same thoughts but Schema.from_pandas(dataframe, preserve_index=False) does not work with Koalas dataframes for some reason

Reply:
Will this be an issue with Koalas since Koalas doesn't support the `category` dtype?

Reply:
Koalas lack of  `category` support is fine since orc doesn't support the `category` dtype either. The problem is that nullable dtypes are parsed differently by `pyarrow.Table.from_pandas` and `pyarrow.Schema.from_pandas`. The schema parses them as null types whereas the table parses them as boolean/integer. The following test fails:

```
from pyarrow import Table, orc, Schema, DictionaryType, field, string
import databricks.koalas as ks

df = pd.DataFrame({'is_registered': pd.Series([pd.NA, False, True, True], dtype='boolean'),
                   'nullable_integer': pd.Series([pd.NA, -36, 49, 64], dtype='Int64')})
dataframe = ks.from_pandas(df)
schema = Schema.from_pandas(dataframe, preserve_index=False)
table_schema = Table.from_pandas(dataframe, preserve_index=False).schema

assert schema == table_schema
```
The workaround for this copy operation is getting kind of tedious. Do we know how frequently people use `to_orc`?

Reply:
to_orc isn't heavily used right now, so we can revisit this if needed (and do a simple implementation now).

Reply:
Created issue #1072